internal/controller/instaslice_controller.go

```diff
@@ -298,24 +344,42 @@ func (r *InstasliceReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 					}
 					updateInstasliceObject.Spec.Allocations[podUuid] = allocations
 					if err := r.Update(ctx, &updateInstasliceObject); err != nil {
-						log.FromContext(ctx).Error(err, "Error updating instaslice allocations")
 						return ctrl.Result{Requeue: true}, nil
 					}
-					pod := r.unGatePod(pod)
-					errForUngating := r.Update(ctx, pod)
-					if errForUngating != nil {
-						log.FromContext(ctx).Error(errForUngating, "failed to ungate pod")
-						return ctrl.Result{Requeue: true}, nil
+					if gpuOperatorPodOk {
+						// Add nodeSelector to the pod
+						if pod.Spec.NodeSelector == nil {
+							pod.Spec.NodeSelector = make(map[string]string)
+						}
+						pod.Spec.NodeSelector[NodeLabel] = allocations.Nodename
+
+						pod := r.unGatePod(pod)
+						errForUngating := r.Update(ctx, pod)
+						if errForUngating != nil {
+							return ctrl.Result{Requeue: true}, nil
+						}
+					} else {
+						log.FromContext(ctx).Info("gpuOperatorPod is not found waiting for it to be in state Running")
+						return ctrl.Result{RequeueAfter: 2 * time.Second}, nil
```

Comment:
There are a bunch of requeueafters... The time (2 * time.Second) should be a constant.

Reply:
Done, thanks